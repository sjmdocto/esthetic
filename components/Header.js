import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';

// TO-DO: Create Filter menu

const EST_ORANGE = 'rgb(227, 131, 4)';
// const EST_ORANGE_TRANSP = 'rgba(227, 131, 4, 0.8)';
const HEADER_BUTTON_SIZE = 35;

const Header = (props) => {
  return (
    <View style={styles.header}>
      <View style={styles.menuButton}>
        <Icon
          name="menu"
          size={HEADER_BUTTON_SIZE}
          color={EST_ORANGE}
          onPress={props.onPressMenu}
        />
      </View>
      <View style={styles.titleContainer}>
        <Text style={styles.titleText}>My Closet</Text>
      </View>
      <Menu style={styles.addButton}>
        <MenuTrigger customStyles={triggerStyles}>
          <Icon name="plus" size={HEADER_BUTTON_SIZE} color={EST_ORANGE} />
        </MenuTrigger>
        <MenuOptions customStyles={optionsStyles}>
          <MenuOption value={1} text="Take photo" />
          <MenuOption value={2} text="Import..." />
        </MenuOptions>
      </Menu>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    // main direction, (which justifyContent works on)
    // is horizontal
    justifyContent: 'center',
    // cross-direction, (which alignItems works on)
    //  is vertical
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: 'black',
    height: '7%',
    flex: 0,
  },
  menuButton: {
    left: -110,
  },
  titleContainer: {
    alignContent: 'center',
  },
  titleText: {
    fontSize: 17,
  },
  addButton: {
    right: -110,
  },
  addMenu: {
    backgroundColor: 'lightgray',
    margin: 50,
  },
});

const optionsStyles = {
  optionsContainer: {
    padding: 1,
  },
  optionsWrapper: {},
};

const triggerStyles = {
  // DOESNT WORK ON ANDROID
  /*
    triggerOuterWrapper: {
        bottom: -30, // eqaul to triggerWrapper.bottom * -1
        right: -50,
    },
    triggerWrapper: {
        bottom: 30, // equal to triggerOuterWrapper.bottom * -1
        right: 50,
    },
    */
};

export default Header;
