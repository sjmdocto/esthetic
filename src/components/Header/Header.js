import React from 'react';
import {View, Text} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';
import {
  HEADER_BUTTON_SIZE,
  styles,
  optionsStyles,
  triggerStyles,
} from './Header.style';
import theme from '../../styles/theme.style';

/**
 * Contains Menu button, title, Add button, and TakePhoto screen modal
 * @param {*} props
 * @param {func} props.openFilterMenu
 * @param {func} props.openCamera
 */
const Header = (props) => {
  return (
    <View style={styles.header}>
      <View style={styles.menuButton}>
        <Icon
          name="menu"
          size={HEADER_BUTTON_SIZE}
          color={theme.EST_ORANGE}
          onPress={props.openFilterMenu}
          accessible={true}
          accessibilityLabel={'Menu button'}
        />
      </View>
      <View style={styles.titleContainer}>
        <Text style={styles.titleText}>My Closet</Text>
      </View>
      <Menu style={styles.addButton}>
        <MenuTrigger
          customStyles={triggerStyles}
          accessible={true}
          accessibilityLabel={'Add to closet button'}>
          <Icon
            name="plus"
            size={HEADER_BUTTON_SIZE}
            color={theme.EST_ORANGE}
          />
        </MenuTrigger>
        <MenuOptions customStyles={optionsStyles}>
          <MenuOption value={1} onSelect={props.openCamera} text="Take photo" />
          <MenuOption value={2} text="Import..." />
        </MenuOptions>
      </Menu>
    </View>
  );
};

export default Header;
