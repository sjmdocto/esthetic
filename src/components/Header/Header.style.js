import {StyleSheet} from 'react-native';
import theme from '../../styles/theme.style';

export const HEADER_BUTTON_SIZE = 35;

export const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    // main direction, (which justifyContent works on)
    // is horizontal
    justifyContent: 'center',
    // cross-direction, (which alignItems works on)
    //  is vertical
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: 'black',
    height: '7%',
    flex: 0,
    backgroundColor: theme.EST_LIGHT_GREY,
  },
  menuButton: {
    left: -110,
  },
  titleContainer: {
    alignContent: 'center',
  },
  titleText: {
    fontSize: 17,
  },
  addButton: {
    right: -110,
  },
});

export const optionsStyles = {
  optionsContainer: {
    padding: 1,
  },
  optionsWrapper: {},
};
