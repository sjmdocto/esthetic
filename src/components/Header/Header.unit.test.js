import React from 'react';
import Header from './Header';
import {MenuProvider} from 'react-native-popup-menu';
import {render, fireEvent} from 'react-native-testing-library';

describe('Header test suite', () => {
  const mockOpenFilterMenu = jest.fn().mockName('openFilterMenu');
  const mockOpenCameraScreen = jest.fn().mockName('openCameraScreen');

  test('Pressing menu button works', () => {
    const {getByLabelText} = render(
      <MenuProvider>
        <Header
          openFilterMenu={mockOpenFilterMenu}
          openCameraScreen={mockOpenCameraScreen}
        />
      </MenuProvider>,
    );
    fireEvent.press(getByLabelText('Menu button'));
    expect(mockOpenFilterMenu).toHaveBeenCalled();
  });

  test('Header renders properly', () => {
    const {toJSON} = render(
      <MenuProvider>
        <Header
          openFilterMenu={mockOpenFilterMenu}
          openCameraScreen={mockOpenCameraScreen}
        />
      </MenuProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  // Can't seem to test if react-native-popup-menu appears after butoon is pressed
  // test('Pressing add button works', () => {
  //   const {getByLabelText, toJSON} = render(
  //     <Header
  //       openFilterMenu={mockOpenFilterMenu}
  //       openCameraScreen={mockOpenCameraScreen}
  //     />,
  //   );
  //   fireEvent.press(getByLabelText('Add to closet button'));
  //   expect(toJSON()).toMatchSnapshot();
  // });
});
