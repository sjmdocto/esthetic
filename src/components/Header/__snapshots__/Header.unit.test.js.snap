// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test suite Header renders properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "row",
          "height": "7%",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "left": -110,
          }
        }
      >
        <Text
          accessibilityLabel="Menu button"
          accessible={true}
          allowFontScaling={false}
          onPress={
            [MockFunction openFilterMenu] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          style={
            Array [
              Object {
                "color": "rgb(227, 131, 4)",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
            }
          }
        >
          My Closet
        </Text>
      </View>
      <View
        style={
          Object {
            "right": -110,
          }
        }
      >
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              accessibilityLabel="Add to closet button"
              accessible={true}
              ctx={
                Object {
                  "menuActions": Object {
                    "_getOpenedMenu": [Function],
                    "_notify": [Function],
                    "closeMenu": [Function],
                    "isMenuOpen": [Function],
                    "openMenu": [Function],
                    "toggleMenu": [Function],
                  },
                  "menuRegistry": Object {
                    "getAll": [Function],
                    "getMenu": [Function],
                    "setOptionsCustomStyles": [Function],
                    "subscribe": [Function],
                    "unsubscribe": [Function],
                    "updateLayoutInfo": [Function],
                  },
                }
              }
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "rgb(227, 131, 4)",
                      "fontSize": 35,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
